### ISSUE TITLE
Rancher CLI: No configuration found after successful login

### ISSUE THREAD

**Issue by shibumi** [rancher/rancher#20288]
**What kind of request is this (question/bug/enhancement/feature request):**

bug

**Steps to reproduce (least amount of steps as possible):**

Download Rancher CLI for Linux from the Rancher Web UI.
Unpack the archive, make the rancher binary executable and run the login:

```
chris akira ~/rancher-v2.2.0 13:19:00 8f929dbd master 
❯ ./rancher l https://rancher1.URL.TLD -t XXXX
chris akira ~/rancher-v2.2.0 13:19:02 8f929dbd master  
❯ echo $?
0
chris akira ~/rancher-v2.2.0 13:19:04 8f929dbd master  
❯ ./rancher nodes
FATA[0000] no configuration found, run 'login' 
```

**Result:**

No configuration is generated after successful login

**Other details that may be helpful:**

**Environment information**
- Rancher version (`rancher/rancher`/`rancher/server` image tag or shown bottom left in the UI): v2.2.2
- Installation option (single install/HA): Multi-Node Cluster in VMWare

**Cluster information**
- Cluster type (Hosted/Infrastructure Provider/Custom/Imported): Self Hosted
- Machine type (cloud/VM/metal) and specifications (CPU/memory): VMs
- Kubernetes version (use `kubectl version`): kubectl inside of rancher cli

**Comment by shibumi** [comment]
I solved the problem by my self.. the reason were missing write permissions in my home directory.
It would be nice if rancher would be more verbose and output a "File permission" error and exit exit code 1

**Comment by dramich** [comment]
Available in v2.3.0-rc3

**Comment by sowmyav27** [comment]
reproduced the issue in `v2.2.0`
Steps:
- docker run --net=host -ti -w /tmp --user 1000:1000 centos bash
- curl -L https://github.com/rancher/cli/releases/download/v2.2.0/rancher-linux-amd64-v2.2.0.tar.gz | tar xvzf - 2>/dev/null
- ./rancher-v2.2.0/rancher login -t token-xxxxx:xxxxxxxxxxxxxxxxxxxxxx https://<server ip>
- type yes to confirm cacerts if self signed
- ./rancher-v2.2.0/rancher projects
- Error `FATA[0000] no configuration found, run `login`  ` is seen

Validated the fix in `v2.3.0-rc3`
- docker run --net=host -ti -w /tmp --user 1000:1000 centos bash
- curl -L https://github.com/rancher/cli/releases/download/v2.2.0/rancher-linux-amd64-v2.2.0.tar.gz | tar xvzf - 2>/dev/null
- ./rancher-v2.2.0/rancher login -t token-xxxxx:xxxxxxxxxxxxxxxxxxxxxx https://<server ip>
- type yes to confirm cacerts if self signed
- Error - `FATA[0002] mkdir /.rancher: permission denied  ` is seen.
